src/renderer/components/+preferences/select-metrics-dialog.tsx

```diff
@@ -0,0 +1,124 @@
+import "./select-metrics-dialog.scss";
+
+import React from "react";
+import { observer } from "mobx-react";
+import { observable } from "mobx";
+import { Dialog, DialogProps } from "../dialog/dialog";
+import { Checkbox } from "../checkbox/checkbox";
+import { Wizard, WizardStep } from "../wizard/wizard";
+import { userStore } from "../../../common/user-store";
+import { Select, SelectOption } from "../select/select";
+import { Icon } from "../icon/icon";
+import { components, PlaceholderProps } from "react-select";
+
+interface Props extends Partial<DialogProps> {
+}
+
+export enum ResourceType {
+  Cluster = "Cluster",
+  Node = "Node",
+  Pod = "Pod",
+  Deployment = "Deployment",
+  StatefulSet = "StatefulSet",
+  Container = "Container",
+  Ingress = "Ingress",
+  VolumeClaim = "VolumeClaim",
+  ReplicaSet = "ReplicaSet",
+  DaemonSet = "DaemonSet",
+}
+
+const Placeholder = observer((props: PlaceholderProps<any>) => {
+  const getPlaceholder = (): React.ReactNode => {
+    if (userStore.hiddenMetrics.size >= 1) {
+      return <>Metrics: {Array.from(userStore.hiddenMetrics).join(", ")}</>;
+    }
+    else {
+      return <>Select metrics...</>;
+    }
+  };
+
+  return (
+    <components.Placeholder {...props}>
+      {getPlaceholder()}
+    </components.Placeholder>
+  );
+});
+
+@observer
+export class SelectMetricsDialog extends React.Component<Props> {
+
+  @observable static isOpen = false;
+
+  static open() {
+    SelectMetricsDialog.isOpen = true;
+  }
+
+  static close() {
+    SelectMetricsDialog.isOpen = false;
+  }
+
+  close = () => {
+    SelectMetricsDialog.close();
+  };
```

Comment:
Do we need a method as well as the static function?

Reply:
I can delete `close` method and change in dialog     
    ```return (
      <Dialog
         .......
        close={() => SelectMetricsDialog.close()}
      >
)```
 Wdyt?

Reply:
```typescript
return (
      <Dialog  
         .........
          close={SelectMetricsDialog.close}
      >
      ......
)
```

Static functions don't need "this" so they are fine to be used in a point-free style

Reply:
thank you. I'll fix